packages/pigeon/lib/src/dart/dart_generator.dart

```diff
@@ -1152,9 +1153,20 @@ if (wrapped == null) {
     }
     returnStatement = '$returnStatement;';
 
+    indent.writeln(
+      'final Future<Object?> ${varNamePrefix}sendFuture = ${varNamePrefix}channel.send($sendArgument);',
+    );
+
+    // If the message call is not made inside of an async method, this creates
+    // an anonymous function to handle the send future.
+    if (!insideAsyncMethod) {
+      indent.writeln('() async {');
+      indent.inc();
+    }
+
     indent.format('''
 final List<Object?>? ${varNamePrefix}replyList =
-\t\tawait ${varNamePrefix}channel.send($sendArgument) as List<Object?>?;
+\t\tawait ${varNamePrefix}sendFuture as List<Object?>?;
```

Comment:
This variable name (`${varNamePrefix}sendFuture`) should be a local variable used in the two places, since it needs to match.